storm-core/src/clj/org/apache/storm/pacemaker/pacemaker.clj

```diff
@@ -0,0 +1,248 @@
+;; Licensed to the Apache Software Foundation (ASF) under one
+;; or more contributor license agreements.  See the NOTICE file
+;; distributed with this work for additional information
+;; regarding copyright ownership.  The ASF licenses this file
+;; to you under the Apache License, Version 2.0 (the
+;; "License"); you may not use this file except in compliance
+;; with the License.  You may obtain a copy of the License at
+;;
+;; http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+
+(ns org.apache.storm.pacemaker.pacemaker
+  (:import [org.apache.storm.pacemaker PacemakerServer IServerMessageHandler]
+           [java.util.concurrent ConcurrentHashMap ThreadPoolExecutor TimeUnit LinkedBlockingDeque]
+           [java.util.concurrent.atomic AtomicInteger]
+           [java.util Date]
+           [backtype.storm.generated
+            HBAuthorizationException HBExecutionException HBNodes HBRecords
+            HBServerMessageType HBMessage HBMessageData HBPulse])
+  (:use [clojure.string :only [replace-first split]]
+        [backtype.storm log config util])
+  (:require [clojure.java.jmx :as jmx])
+  (:gen-class))
+
+;; This is the old Thrift service that this server is emulating.
```

Comment:
Its not really old.  And can you explain why you are emulating a thrift service instead of using thrift?